### PR TITLE
Simplify component naming

### PR DESCRIPTION
**Pull request by GeorgesLorre**
This fixes the inconsistent folder names between runs (cached vs non-cached) and cleans up the different component/folder naming logic we have.

### CODE REVIEW

File: src/fondant/core/component_spec.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def __init__(
         tags: t.Optional[t.List[str]] = None,
     ):
         spec_dict: t.Dict[str, t.Any] = {
-            "name": name,
+            "name": self.sanitized_component_name(name),
```
Comment (GeorgesLorre) [inline src/fondant/core/component_spec.py:104]: This should be the only place where the incoming name is cleaned and stored

File: src/fondant/pipeline/pipeline.py [hunk 2]
```diff
@@ -305,6 +300,10 @@ def _get_registry_path(name: str) -> Path:
             raise ValueError(msg)
         return component_dir
 
+    @property
+    def component_id(self) -> str:
```
Comment (RobbeSneyders) [inline src/fondant/pipeline/pipeline.py:304]: Is there a reason to call this `id` here instead of `name`?

Comment (GeorgesLorre) [reply src/fondant/pipeline/pipeline.py:304]: I'll update it

File: src/fondant/pipeline/compiler.py [hunk 3]
```diff
@@ -353,48 +354,55 @@ def from_fondant_component_spec(
             },
         }
 
-        cleaned_component_name = fondant_component.sanitized_component_name
-
+        kfp_safe_name = (
```
Comment (mrchtr) [inline src/fondant/pipeline/compiler.py:357]: Nice name! Makes it clear.

File: src/fondant/pipeline/compiler.py [hunk 4]
```diff
@@ -353,48 +354,55 @@ def from_fondant_component_spec(
...
-                            "taskInfo": {"name": cleaned_component_name},
+                            "taskInfo": {"name": kfp_safe_name},
                         },
                     },
                 },
                 "inputDefinitions": input_definitions,
             },
             "schemaVersion": "2.1.0",
-            "sdkVersion": "kfp-2.0.1",
+            "sdkVersion": "kfp-2.6.0",
```
Comment (mrchtr) [inline src/fondant/pipeline/compiler.py:405]: Any side effects by increasing the version?

**Review by mrchtr — APPROVED** [review]
Thanks @GeorgesLorre!

File: src/fondant/pipeline/compiler.py (hunk 4 above)
Comment (GeorgesLorre) [reply src/fondant/pipeline/compiler.py:405]: No, now it is in line with the KFP version we use now.

**Pull request merged**